src/main/java/org/apache/sysds/hops/ReorgOp.java

```diff
@@ -76,6 +77,8 @@ public ReorgOp(String l, DataType dt, ValueType vt, ReOrgOp o, ArrayList<Hop> in
 			getInput().add(i, in);
 			in.getParent().add(this);
 		}
+
+		updateETFed();
```

Comment:
double updateETFed here maybe only one needed?

Reply:
It is important to have both. In the first one, the execution type and fedoutput will be set during the construction of the Hops (in the order input to output). In the second call in "optFindExecType", it will set the exectype so that a once federated execution is not converted to a CP. This will happen in the order output to input (plus additional calls in certain cases).